annotations/utils.py

```diff
@@ -37,6 +38,114 @@ def basepath(request):
+            "creationTime": creation_time.strftime('%Y-%m-%d'),
+            "creationPlace": "",
+            "sourceUri": source_uri
+        }
+    }
+
+def get_relation_node(user, creation_time, source_uri):
+    """
+    Helper function to build a relation node.
+    """
+    return {
+        "label": "",
+        "metadata": {
+            "type": "relation_event"
+        },
+        "context": {
+            "creator": user.username,
+            "creationTime": creation_time.strftime('%Y-%m-%d'),
+            "creationPlace": "",
+            "sourceUri": source_uri
+        }
+    }
+
+def generate_graph_data(relationset, user):
+    nodes = {}
+    edges = []
+    node_counter = 0
+    source_uri = relationset.occursIn.uri
+
+    def get_node_id():
+        nonlocal node_counter
+        node_id = str(node_counter)
+        node_counter += 1
+        return node_id
+
+    for relation in relationset.constituents.all():
+        subject = getattr(relation.source_content_object, 'interpretation', None)
+        obj = getattr(relation.object_content_object, 'interpretation', None)
+        predicate = getattr(relation.predicate, 'interpretation', None)
+        
+        # Extract the creation time from the appellation
+        creation_time = relationset.occursIn.created
+
+        if subject and subject.uri not in nodes:
+            subject_id = get_node_id()
+            nodes[subject_id] = build_concept_node(subject, user, creation_time, source_uri)
+
+        if obj and obj.uri not in nodes:
+            obj_id = get_node_id()
+            nodes[obj_id] = build_concept_node(obj, user, creation_time, source_uri)
+
+        if predicate and predicate.uri not in nodes:
+            predicate_id = get_node_id()
+            nodes[predicate_id] = build_concept_node(predicate, user, creation_time, source_uri)
+
+        relation_id = get_node_id()
+        nodes[relation_id] = get_relation_node(user, creation_time, source_uri)
+
+        if subject and predicate:
+            edges.append({"source": subject_id, "relation": "subject", "target": predicate_id})
+
+        if predicate and obj:
+            edges.append({"source": predicate_id, "relation": "predicate", "target": obj_id})
+
+        edges.append({"source": relation_id, "relation": "object", "target": obj_id})
+
+    return {
+        "graph": {
+            "metadata": {
+                "defaultMapping": {
+                    "subject": {"type": "REF", "reference": "0"},
+                    "predicate": {"type": "URI", "uri": "", "label": ""},
+                    "object": {"type": "REF", "reference": "3"}
+                },
+                "context": {
+                    "creator": user.username,
+                    "creationTime": timezone.now().strftime('%Y-%m-%d'),
+                    "creationPlace": "",
+                    "sourceUri": source_uri
+                }
+            },
+            "nodes": nodes,
+            "edges": edges
+        }
+    }
+
+
```

Comment:
these two functions are specific to the quadriga api, aren't they? They should not be in the annotations util file.